src/full/Agda/Syntax/Abstract/Name.hs

```diff
@@ -386,7 +389,11 @@ instance Pretty ModuleName where
   pretty = hcat . punctuate "." . map pretty . mnameToList
 
 instance Pretty QName where
-  pretty = hcat . punctuate "." . map pretty . qnameToList0
+  pretty = hcat . punctuate "." . map pretty . qnameToList0 . useCanonical
```

Comment:
I wonder whether it hadn't been easier to unfold `qnameToList` here (is a one-liner).